### PR TITLE
transfer quantity matchers from SIMONA

### LINKED ISSUE #361: Transfer SIMONAs QuantityMatchers
We have started a QuantityMatcher implementation in the `PowerSystemUtils` maybe we should also port them here. We could mark them as deprecated and leave a comment that we want to reference the PSU implementation in the future. And also open a new PSU issue to copy the functionality.

_Originally posted by @t-ober in https://github.com/ie3-institute/simona/pull/82#discussion_r1187244582_

### PR DESCRIPTION
**Pull request by danielfeismann**
resolves #361

### CODE REVIEW

**Comment by sebastian-peter** [comment]
I have one critical remark here: These matchers are supposed to be solely used in tests. Here, they are now transferred into "main" code, which is also available e.g. in SIMONA, where the dependency is added via `implementation` in gradle (same for any other project where PSU are not added _just_ for tests, i.e. practically all).
This means, that when compiling SIMONA, these classes are compiled and kept with all other "main" code as well, which they technically shouldn't be. Code that is only needed for tests does not have to be available in main. 

So I'm not fundamentally opposed to making the quantity matchers more universally available, but this might be something to consider. Also, since we're moving towards using squants more, this might not be a something that we absolutely need?

**Comment by sebastian-peter** [comment]
Also a reminder that comparison matching with quantities can be done via `a should be < b`, `a should be <= b` etc as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 - Added quantity for volumetric flow rate [#363](https://github.com/ie3-institute/PowerSystemUtils/issues/363)
+-   Transfer QuantityMatchers from SIMONA [#361](https://github.com/ie3-institute/PowerSystemUtils/issues/361)
```
Comment (jo-bao) [inline CHANGELOG.md:26]: There is a tab too much.

File: src/main/scala/edu/ie3/util/quantities/QuantityMatchers.scala [hunk 2]
```diff
@@ -38,4 +42,115 @@ case object QuantityMatchers extends QuantityMatchers {
...
+    * @param left
+    *   Left hand side of the comparison
+    * @param right
+    *   Right hand side of the comparison
+    * @param tolerance
+    *   Numerical tolerance to be applied to the right hand side
+    * @param rightWithToleranceFun
+    *   Function, how to build the right hand side with tolerance
+    * @param compareFun
+    *   Compare function (lhs, rhs) => Condition
```
Comment (jo-bao) [inline src/main/scala/edu/ie3/util/quantities/QuantityMatchers.scala:117]: I am not sure, if I would call it "Condition"

**Review by jo-bao — CHANGES_REQUESTED** [review]
Overall it looks good to me! Just a few minor points.

File: CHANGELOG.md (hunk 1 above)
Comment (danielfeismann) [reply CHANGELOG.md:26]: Good catch!

**Comment by danielfeismann** [comment]
Since with [this commit](https://github.com/ie3-institute/simona/pull/82/commits/ef5adeec9da14d8ffa2e31338eb8db98f382e8d0) the proposed changes are no longer direclty needed in SIMONA we decided to close this PR.

**Pull request closed without merging**